datatypes/src/raster/operations/classification.rs

```diff
@@ -0,0 +1,84 @@
+use crate::raster::{data_type::FromPrimitive, BaseRaster, GridDimension, Pixel, Raster};
+use crate::util::Result;
+use num_traits::AsPrimitive;
+
+pub trait RasterClassification<D, C>
+where
+    D: GridDimension,
+{
+    type Output: Raster<D, u32, C>;
+    fn classification(
+        &self,
+        bounds_class: &[(f64, f64, u32)],
+        no_data_class: u32,
+    ) -> Result<Self::Output>;
+}
+
+impl<D, T> RasterClassification<D, Vec<u32>> for BaseRaster<D, T, Vec<T>>
+where
+    D: GridDimension + Copy,
+    T: Pixel + FromPrimitive<f64> + AsPrimitive<f64>,
+{
+    type Output = BaseRaster<D, u32, Vec<u32>>;
+    fn classification(
+        &self,
+        bounds_class: &[(f64, f64, u32)],
+        no_data_class: u32,
+    ) -> Result<Self::Output> {
+        let pixels: Vec<u32> = self
+            .data_container
+            .iter()
+            .map(|&p| {
+                if let Some(no_data_value) = self.no_data_value {
+                    if p == no_data_value {
+                        return no_data_class;
+                    }
+                }
+
+                let class = bounds_class.iter().find_map(|&(lower, upper, class)| {
+                    let p_f64: f64 = p.as_();
+                    if (p_f64 >= lower) && (p_f64 <= upper) {
+                        return Some(class);
+                    }
+                    None
+                });
+                class.unwrap_or(no_data_class)
+            })
+            .collect();
+
+        BaseRaster::new(
```

Comment:
The classification bounds and labels need to be stored like in DataDescription in mapping